### PR TITLE
DOC: fix some doctest failures

### LINKED ISSUE #12483: DOC, TST: matmul docstring fails refguide
The matmul docstring in `numpy/core/code_generators/ufunc_docstrings.py` fails the refguide check in #12253 for a variety of reasons varying from behavior changes to not even closing brackets in `np.array()` declarations.

While I've cleaned up some of the trivial issues, I'm going to mark the rest of the issues in that docstring with the `SKIP` directive and I suggest that they be fixed **after** that PR is merged so that the fixes are properly verified in CI.

I'll probably continue with this approach moving forward until the doctest PR is merged -- adding SKIPs and opening bookmark issues--otherwise I end up as the "human refguide checker" for doc changes.

### PR DESCRIPTION
**Pull request by mattip**
Fixes #12483 by updating the docstring tests in matmul. Also tweak the exception message for other doctests

### CODE REVIEW

File: numpy/core/code_generators/ufunc_docstrings.py [hunk 1]
```diff
@@ -846,7 +846,7 @@ def add_newdoc(place, name, doc):
     >>> np.cos(np.zeros((3,3)),np.zeros((2,2)))
     Traceback (most recent call last):
       File "<stdin>", line 1, in <module>
-    ValueError: invalid return array shape
+    ValueError: operands could not be broadcast together with shapes (3,3) (2,2)
```
[the remark this answers is not in the document]
Comment (mattip) [reply numpy/core/code_generators/ufunc_docstrings.py]: These results *are* being hit the module is being tested. I discovered these tweaks by putting a pdb in `refguide_check.py` where `check_output` returned `False`. It seems doctest is not particular about the error message, so I just replaced `wanted` with `got`

**Review by tylerjereddy — APPROVED** [review]
This should be fine to merge then. If people feel passionately that refguide should fail on changing traceback error messages (I suspect it is only sensitive to the type of error), they could bring that up in the refguide wishlist issue.

Might have been nice to briefly indicate which docstring(s) are fixed in the commit message I suppose.

**Comment by charris** [comment]
Thanks Matti. @tylerjereddy You should just go ahead and merge straight forward stuff like this.

**Comment by mattip** [comment]
Marking this for backporting. See #13355

**Pull request merged**